File: antlr/src/main/java/com/rescribe/antlr/parse/schema/NestedObject.java
package com.rescribe.antlr.parse.schema;

import lombok.*;
import lombok.experimental.SuperBuilder;
import org.bson.types.ObjectId;

@AllArgsConstructor
@Data
@SuperBuilder
public class NestedObject {
  final String _id;
  Parent parent;
  Location location;

  public NestedObject() {
    this._id = new ObjectId().toHexString();
  }
}

File: api/src/schema/antlr/nestedObject.ts
import { ObjectType, Field } from 'type-graphql';
import Location from './location';
import { ObjectId } from 'mongodb';
import Parent from './parent';

@ObjectType({description: 'class'})
export default class NestedObject {
  @Field()
  readonly _id: ObjectId;
  @Field(_type => Parent, { description: 'parent' })
  parent: Parent;
  @Field(_type => Location, { description: 'location' })
  location: Location;
}
